// @sapphire/framework/dist/lib/types/Enums.js
'use strict';

// src/lib/types/Enums.ts
var CooldownLevel = /* @__PURE__ */ ((CooldownLevel2) => {
  CooldownLevel2["Author"] = "author";
  CooldownLevel2["Channel"] = "channel";
  CooldownLevel2["Guild"] = "guild";
  return CooldownLevel2;
})(CooldownLevel || {});
var PluginHook = /* @__PURE__ */ ((PluginHook2) => {
  PluginHook2["PreGenericsInitialization"] = "preGenericsInitialization";
  PluginHook2["PreInitialization"] = "preInitialization";
  PluginHook2["PostInitialization"] = "postInitialization";
  PluginHook2["PreLogin"] = "preLogin";
  PluginHook2["PostLogin"] = "postLogin";
  return PluginHook2;
})(PluginHook || {});
var BucketScope = /* @__PURE__ */ ((BucketScope2) => {
  BucketScope2[BucketScope2["Channel"] = 0] = "Channel";
  BucketScope2[BucketScope2["Global"] = 1] = "Global";
  BucketScope2[BucketScope2["Guild"] = 2] = "Guild";
  BucketScope2[BucketScope2["User"] = 3] = "User";
  return BucketScope2;
})(BucketScope || {});
var RegisterBehavior = /* @__PURE__ */ ((RegisterBehavior2) => {
  RegisterBehavior2["Overwrite"] = "OVERWRITE";
  RegisterBehavior2["LogToConsole"] = "LOG_TO_CONSOLE";
  RegisterBehavior2["VerboseOverwrite"] = "VERBOSE_OVERWRITE";
  return RegisterBehavior2;
})(RegisterBehavior || {});
var InternalRegistryAPIType = /* @__PURE__ */ ((InternalRegistryAPIType2) => {
  InternalRegistryAPIType2[InternalRegistryAPIType2["ChatInput"] = 0] = "ChatInput";
  InternalRegistryAPIType2[InternalRegistryAPIType2["ContextMenu"] = 1] = "ContextMenu";
  return InternalRegistryAPIType2;
})(InternalRegistryAPIType || {});

exports.BucketScope = BucketScope;
exports.CooldownLevel = CooldownLevel;
exports.InternalRegistryAPIType = InternalRegistryAPIType;
exports.PluginHook = PluginHook;
exports.RegisterBehavior = RegisterBehavior;
