// @sapphire/framework/dist/lib/types/Events.js
'use strict';

var discord_js = require('discord.js');

// src/lib/types/Events.ts
var Events = {
  ChannelCreate: discord_js.Constants.Events.CHANNEL_CREATE,
  ChannelDelete: discord_js.Constants.Events.CHANNEL_DELETE,
  ChannelPinsUpdate: discord_js.Constants.Events.CHANNEL_PINS_UPDATE,
  ChannelUpdate: discord_js.Constants.Events.CHANNEL_UPDATE,
  ClientReady: discord_js.Constants.Events.CLIENT_READY,
  Debug: discord_js.Constants.Events.DEBUG,
  Error: discord_js.Constants.Events.ERROR,
  GuildBanAdd: discord_js.Constants.Events.GUILD_BAN_ADD,
  GuildBanRemove: discord_js.Constants.Events.GUILD_BAN_REMOVE,
  GuildCreate: discord_js.Constants.Events.GUILD_CREATE,
  GuildDelete: discord_js.Constants.Events.GUILD_DELETE,
  GuildEmojiCreate: discord_js.Constants.Events.GUILD_EMOJI_CREATE,
  GuildEmojiDelete: discord_js.Constants.Events.GUILD_EMOJI_DELETE,
  GuildEmojiUpdate: discord_js.Constants.Events.GUILD_EMOJI_UPDATE,
  GuildIntegrationsUpdate: discord_js.Constants.Events.GUILD_INTEGRATIONS_UPDATE,
  GuildMemberAdd: discord_js.Constants.Events.GUILD_MEMBER_ADD,
  GuildMemberAvailable: discord_js.Constants.Events.GUILD_MEMBER_AVAILABLE,
  GuildMemberRemove: discord_js.Constants.Events.GUILD_MEMBER_REMOVE,
  GuildMembersChunk: discord_js.Constants.Events.GUILD_MEMBERS_CHUNK,
  GuildMemberUpdate: discord_js.Constants.Events.GUILD_MEMBER_UPDATE,
  GuildRoleCreate: discord_js.Constants.Events.GUILD_ROLE_CREATE,
  GuildRoleDelete: discord_js.Constants.Events.GUILD_ROLE_DELETE,
  GuildRoleUpdate: discord_js.Constants.Events.GUILD_ROLE_UPDATE,
  GuildStickerCreate: discord_js.Constants.Events.GUILD_STICKER_CREATE,
  GuildStickerDelete: discord_js.Constants.Events.GUILD_STICKER_DELETE,
  GuildStickerUpdate: discord_js.Constants.Events.GUILD_STICKER_UPDATE,
  GuildUnavailable: discord_js.Constants.Events.GUILD_UNAVAILABLE,
  GuildUpdate: discord_js.Constants.Events.GUILD_UPDATE,
  InteractionCreate: discord_js.Constants.Events.INTERACTION_CREATE,
  Invalidated: discord_js.Constants.Events.INVALIDATED,
  InvalidRequestWarning: discord_js.Constants.Events.INVALID_REQUEST_WARNING,
  InviteCreate: discord_js.Constants.Events.INVITE_CREATE,
  InviteDelete: discord_js.Constants.Events.INVITE_DELETE,
  MessageBulkDelete: discord_js.Constants.Events.MESSAGE_BULK_DELETE,
  MessageCreate: discord_js.Constants.Events.MESSAGE_CREATE,
  MessageDelete: discord_js.Constants.Events.MESSAGE_DELETE,
  MessageReactionAdd: discord_js.Constants.Events.MESSAGE_REACTION_ADD,
  MessageReactionRemove: discord_js.Constants.Events.MESSAGE_REACTION_REMOVE,
  MessageReactionRemoveAll: discord_js.Constants.Events.MESSAGE_REACTION_REMOVE_ALL,
  MessageReactionRemoveEmoji: discord_js.Constants.Events.MESSAGE_REACTION_REMOVE_EMOJI,
  MessageUpdate: discord_js.Constants.Events.MESSAGE_UPDATE,
  PresenceUpdate: discord_js.Constants.Events.PRESENCE_UPDATE,
  RateLimit: discord_js.Constants.Events.RATE_LIMIT,
  Raw: discord_js.Constants.Events.RAW,
  ShardDisconnect: discord_js.Constants.Events.SHARD_DISCONNECT,
  ShardError: discord_js.Constants.Events.SHARD_ERROR,
  ShardReady: discord_js.Constants.Events.SHARD_READY,
  ShardReconnecting: discord_js.Constants.Events.SHARD_RECONNECTING,
  ShardResume: discord_js.Constants.Events.SHARD_RESUME,
  StageInstanceCreate: discord_js.Constants.Events.STAGE_INSTANCE_CREATE,
  StageInstanceDelete: discord_js.Constants.Events.STAGE_INSTANCE_DELETE,
  StageInstanceUpdate: discord_js.Constants.Events.STAGE_INSTANCE_UPDATE,
  ThreadCreate: discord_js.Constants.Events.THREAD_CREATE,
  ThreadDelete: discord_js.Constants.Events.THREAD_DELETE,
  ThreadListSync: discord_js.Constants.Events.THREAD_LIST_SYNC,
  ThreadMembersUpdate: discord_js.Constants.Events.THREAD_MEMBERS_UPDATE,
  ThreadMemberUpdate: discord_js.Constants.Events.THREAD_MEMBER_UPDATE,
  ThreadUpdate: discord_js.Constants.Events.THREAD_UPDATE,
  TypingStart: discord_js.Constants.Events.TYPING_START,
  UserUpdate: discord_js.Constants.Events.USER_UPDATE,
  VoiceServerUpdate: discord_js.Constants.Events.VOICE_SERVER_UPDATE,
  VoiceStateUpdate: discord_js.Constants.Events.VOICE_STATE_UPDATE,
  Warn: discord_js.Constants.Events.WARN,
  WebhooksUpdate: discord_js.Constants.Events.WEBHOOKS_UPDATE,
  PreMessageParsed: "preMessageParsed",
  MentionPrefixOnly: "mentionPrefixOnly",
  NonPrefixedMessage: "nonPrefixedMessage",
  PrefixedMessage: "prefixedMessage",
  UnknownMessageCommandName: "unknownMessageCommandName",
  UnknownMessageCommand: "unknownMessageCommand",
  CommandDoesNotHaveMessageCommandHandler: "commandDoesNotHaveMessageCommandHandler",
  PreMessageCommandRun: "preMessageCommandRun",
  MessageCommandDenied: "messageCommandDenied",
  MessageCommandAccepted: "messageCommandAccepted",
  MessageCommandRun: "messageCommandRun",
  MessageCommandSuccess: "messageCommandSuccess",
  MessageCommandError: "messageCommandError",
  MessageCommandFinish: "messageCommandFinish",
  MessageCommandTypingError: "messageCommandTypingError",
  ListenerError: "listenerError",
  CommandApplicationCommandRegistryError: "commandApplicationCommandRegistryError",
  PiecePostLoad: "piecePostLoad",
  PieceUnload: "pieceUnload",
  PluginLoaded: "pluginLoaded",
  InteractionHandlerParseError: "interactionHandlerParseError",
  InteractionHandlerError: "interactionHandlerError",
  PossibleAutocompleteInteraction: "possibleAutocompleteInteraction",
  CommandAutocompleteInteractionSuccess: "commandAutocompleteInteractionSuccess",
  CommandAutocompleteInteractionError: "commandAutocompleteInteractionError",
  PossibleChatInputCommand: "possibleChatInputCommand",
  UnknownChatInputCommand: "unknownChatInputCommand",
  CommandDoesNotHaveChatInputCommandHandler: "commandDoesNotHaveChatInputCommandHandler",
  PreChatInputCommandRun: "preChatInputCommandRun",
  ChatInputCommandDenied: "chatInputCommandDenied",
  ChatInputCommandAccepted: "chatInputCommandAccepted",
  ChatInputCommandRun: "chatInputCommandRun",
  ChatInputCommandSuccess: "chatInputCommandSuccess",
  ChatInputCommandError: "chatInputCommandError",
  ChatInputCommandFinish: "chatInputCommandFinish",
  PossibleContextMenuCommand: "possibleContextMenuCommand",
  UnknownContextMenuCommand: "unknownContextMenuCommand",
  CommandDoesNotHaveContextMenuCommandHandler: "commandDoesNotHaveContextMenuCommandHandler",
  PreContextMenuCommandRun: "preContextMenuCommandRun",
  ContextMenuCommandDenied: "contextMenuCommandDenied",
  ContextMenuCommandAccepted: "contextMenuCommandAccepted",
  ContextMenuCommandRun: "contextMenuCommandRun",
  ContextMenuCommandSuccess: "contextMenuCommandSuccess",
  ContextMenuCommandError: "contextMenuCommandError",
  ContextMenuCommandFinish: "contextMenuCommandFinish"
};

exports.Events = Events;
