// @sapphire/framework/dist/arguments/CoreGuildThreadChannel.js
'use strict';

var Identifiers_js = require('../lib/errors/Identifiers.js');
var guildThreadChannel_js = require('../lib/resolvers/guildThreadChannel.js');
var Argument_js = require('../lib/structures/Argument.js');

var __defProp = Object.defineProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var CoreArgument = class extends Argument_js.Argument {
  constructor(context) {
    super(context, { name: "guildThreadChannel" });
  }
  run(parameter, context) {
    const { guild } = context.message;
    if (!guild) {
      return this.error({
        parameter,
        identifier: Identifiers_js.Identifiers.ArgumentGuildChannelMissingGuildError,
        message: "This command can only be used in a server.",
        context
      });
    }
    const resolved = guildThreadChannel_js.resolveGuildThreadChannel(parameter, guild);
    return resolved.mapErrInto(
      (identifier) => this.error({
        parameter,
        identifier,
        message: "The given argument did not resolve to a valid thread.",
        context: { ...context, guild }
      })
    );
  }
};
__name(CoreArgument, "CoreArgument");

exports.CoreArgument = CoreArgument;
