// @sapphire/framework/dist/lib/structures/InteractionHandler.js
'use strict';

var pieces = require('@sapphire/pieces');
var result = require('@sapphire/result');

var __defProp = Object.defineProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var InteractionHandler = class extends pieces.Piece {
  constructor(context, options) {
    super(context, options);
    this.interactionHandlerType = options.interactionHandlerType;
  }
  parse(_interaction) {
    return this.some();
  }
  some(data) {
    return result.Option.some(data);
  }
  none() {
    return result.Option.none;
  }
  toJSON() {
    return {
      ...super.toJSON(),
      interactionHandlerType: this.interactionHandlerType
    };
  }
};
__name(InteractionHandler, "InteractionHandler");
var InteractionHandlerTypes = /* @__PURE__ */ ((InteractionHandlerTypes2) => {
  InteractionHandlerTypes2["Button"] = "BUTTON";
  InteractionHandlerTypes2["SelectMenu"] = "SELECT_MENU";
  InteractionHandlerTypes2["ModalSubmit"] = "MODAL_SUBMIT";
  InteractionHandlerTypes2["MessageComponent"] = "MESSAGE_COMPONENT";
  InteractionHandlerTypes2["Autocomplete"] = "AUTOCOMPLETE";
  return InteractionHandlerTypes2;
})(InteractionHandlerTypes || {});

exports.InteractionHandler = InteractionHandler;
exports.InteractionHandlerTypes = InteractionHandlerTypes;
