// @sapphire/framework/dist/preconditions/GuildTextOnly.js
'use strict';

var Identifiers_js = require('../lib/errors/Identifiers.js');
var Precondition_js = require('../lib/structures/Precondition.js');

var __defProp = Object.defineProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var CorePrecondition = class extends Precondition_js.AllFlowsPrecondition {
  constructor() {
    super(...arguments);
    this.allowedTypes = ["GUILD_TEXT", "GUILD_PUBLIC_THREAD", "GUILD_PRIVATE_THREAD"];
  }
  messageRun(message) {
    return this.allowedTypes.includes(message.channel.type) ? this.ok() : this.error({
      identifier: Identifiers_js.Identifiers.PreconditionGuildTextOnly,
      message: "You can only run this message command in server text channels."
    });
  }
  async chatInputRun(interaction) {
    const channel = await this.fetchChannelFromInteraction(interaction);
    return this.allowedTypes.includes(channel.type) ? this.ok() : this.error({
      identifier: Identifiers_js.Identifiers.PreconditionGuildTextOnly,
      message: "You can only run this chat input command in server text channels."
    });
  }
  async contextMenuRun(interaction) {
    const channel = await this.fetchChannelFromInteraction(interaction);
    return this.allowedTypes.includes(channel.type) ? this.ok() : this.error({
      identifier: Identifiers_js.Identifiers.PreconditionGuildTextOnly,
      message: "You can only run this context menu command in server text channels."
    });
  }
};
__name(CorePrecondition, "CorePrecondition");

exports.CorePrecondition = CorePrecondition;
