// @sapphire/framework/dist/preconditions/GuildPublicThreadOnly.js
'use strict';

var Identifiers_js = require('../lib/errors/Identifiers.js');
var Precondition_js = require('../lib/structures/Precondition.js');

var __defProp = Object.defineProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var CorePrecondition = class extends Precondition_js.AllFlowsPrecondition {
  messageRun(message) {
    return message.thread?.type === "GUILD_PUBLIC_THREAD" ? this.ok() : this.error({
      identifier: Identifiers_js.Identifiers.PreconditionGuildPublicThreadOnly,
      message: "You can only run this message command in public server thread channels."
    });
  }
  async chatInputRun(interaction) {
    const channel = await this.fetchChannelFromInteraction(interaction);
    return channel.type === "GUILD_PUBLIC_THREAD" ? this.ok() : this.error({
      identifier: Identifiers_js.Identifiers.PreconditionGuildPublicThreadOnly,
      message: "You can only run this chat input command in public server thread channels."
    });
  }
  async contextMenuRun(interaction) {
    const channel = await this.fetchChannelFromInteraction(interaction);
    return channel.type === "GUILD_PUBLIC_THREAD" ? this.ok() : this.error({
      identifier: Identifiers_js.Identifiers.PreconditionGuildPublicThreadOnly,
      message: "You can only run this context menu command in public server thread channels."
    });
  }
};
__name(CorePrecondition, "CorePrecondition");

exports.CorePrecondition = CorePrecondition;
