// @sapphire/utilities/dist/lib/sleep.mjs
import { __name, __publicField } from '../chunk-AZHX2MBF.mjs';

// src/lib/sleep.ts
var AbortError = class extends Error {
  constructor(message, options) {
    super(message, options);
    __publicField(this, "code");
    this.name = "AbortError";
    this.code = "ERR_ABORT";
  }
};
__name(AbortError, "AbortError");
function sleep(ms, value, options) {
  return new Promise((resolve, reject) => {
    const timer = setTimeout(() => resolve(value), ms);
    const signal = options == null ? void 0 : options.signal;
    if (signal) {
      signal.addEventListener("abort", () => {
        clearTimeout(timer);
        reject(
          new AbortError("The operation was aborted", {
            cause: signal.reason
          })
        );
      });
    }
    if ((options == null ? void 0 : options.ref) === false && typeof timer === "object") {
      timer.unref();
    }
  });
}
__name(sleep, "sleep");
function sleepSync(ms, value) {
  return value;
}
__name(sleepSync, "sleepSync");

export { AbortError, sleep, sleepSync };
