// @sapphire/framework/dist/lib/utils/application-commands/normalizeInputs.js
'use strict';

var builders = require('@discordjs/builders');
var utilities = require('@sapphire/utilities');
var v10 = require('discord-api-types/v10');
var discord_js = require('discord.js');

var __defProp = Object.defineProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
function isBuilder(command) {
  return command instanceof builders.SlashCommandBuilder;
}
__name(isBuilder, "isBuilder");
function addDefaultsToChatInputJSON(data) {
  data.default_permission ?? (data.default_permission = true);
  data.dm_permission ?? (data.dm_permission = true);
  data.type ?? (data.type = v10.ApplicationCommandType.ChatInput);
  data.name_localizations ?? (data.name_localizations = null);
  data.description_localizations ?? (data.description_localizations = null);
  return data;
}
__name(addDefaultsToChatInputJSON, "addDefaultsToChatInputJSON");
function addDefaultsToContextMenuJSON(data) {
  data.default_permission ?? (data.default_permission = true);
  data.dm_permission ?? (data.dm_permission = true);
  data.name_localizations ?? (data.name_localizations = null);
  data.description_localizations ?? (data.description_localizations = null);
  return data;
}
__name(addDefaultsToContextMenuJSON, "addDefaultsToContextMenuJSON");
function normalizeChatInputCommand(command) {
  if (utilities.isFunction(command)) {
    const builder = new builders.SlashCommandBuilder();
    command(builder);
    return addDefaultsToChatInputJSON(builder.toJSON());
  }
  if (isBuilder(command)) {
    return addDefaultsToChatInputJSON(command.toJSON());
  }
  const finalObject = {
    name: command.name,
    name_localizations: command.nameLocalizations,
    description: command.description,
    description_localizations: command.descriptionLocalizations,
    default_permission: command.defaultPermission,
    type: v10.ApplicationCommandType.ChatInput,
    dm_permission: command.dmPermission
  };
  if (command.defaultMemberPermissions) {
    finalObject.default_member_permissions = String(command.defaultMemberPermissions);
  }
  if (command.options?.length) {
    finalObject.options = command.options.map((option) => discord_js.ApplicationCommand["transformOption"](option));
  }
  return addDefaultsToChatInputJSON(finalObject);
}
__name(normalizeChatInputCommand, "normalizeChatInputCommand");
function normalizeContextMenuCommand(command) {
  if (utilities.isFunction(command)) {
    const builder = new builders.ContextMenuCommandBuilder();
    command(builder);
    return addDefaultsToContextMenuJSON(builder.toJSON());
  }
  if (command instanceof builders.ContextMenuCommandBuilder) {
    return addDefaultsToContextMenuJSON(command.toJSON());
  }
  let type;
  switch (command.type) {
    case discord_js.Constants.ApplicationCommandTypes.MESSAGE:
    case "MESSAGE":
      type = v10.ApplicationCommandType.Message;
      break;
    case discord_js.Constants.ApplicationCommandTypes.USER:
    case "USER":
      type = v10.ApplicationCommandType.User;
      break;
    default:
      throw new Error(`Unhandled command type: ${command.type}`);
  }
  const finalObject = {
    name: command.name,
    name_localizations: command.nameLocalizations,
    type,
    default_permission: command.defaultPermission,
    dm_permission: command.dmPermission
  };
  if (command.defaultMemberPermissions) {
    finalObject.default_member_permissions = String(command.defaultMemberPermissions);
  }
  return addDefaultsToContextMenuJSON(finalObject);
}
__name(normalizeContextMenuCommand, "normalizeContextMenuCommand");
function convertApplicationCommandToApiData(command) {
  const returnData = {
    name: command.name,
    name_localizations: command.nameLocalizations,
    default_permission: command.defaultPermission,
    dm_permission: command.dmPermission
  };
  if (command.defaultMemberPermissions) {
    returnData.default_member_permissions = command.defaultMemberPermissions.bitfield.toString();
  }
  if (command.type === "CHAT_INPUT") {
    returnData.type = v10.ApplicationCommandType.ChatInput;
    returnData.description = command.description;
    returnData.description_localizations = command.descriptionLocalizations;
  } else if (command.type === "MESSAGE") {
    returnData.type = v10.ApplicationCommandType.Message;
  } else if (command.type === "USER") {
    returnData.type = v10.ApplicationCommandType.User;
  }
  if (command.options.length) {
    returnData.options = command.options.map((option) => discord_js.ApplicationCommand["transformOption"](option));
  }
  return returnData;
}
__name(convertApplicationCommandToApiData, "convertApplicationCommandToApiData");

exports.convertApplicationCommandToApiData = convertApplicationCommandToApiData;
exports.normalizeChatInputCommand = normalizeChatInputCommand;
exports.normalizeContextMenuCommand = normalizeContextMenuCommand;
